control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -393,23 +397,57 @@ func (r *HostedControlPlaneReconciler) ensureInfrastructure(ctx context.Context,
 		return status, fmt.Errorf("failed to create oauth server route: %w", err)
 	}
 
-	apiAddress, err := getLoadBalancerServiceAddress(r, ctx, client.ObjectKeyFromObject(apiService))
-	if err != nil {
-		return status, fmt.Errorf("failed to get service: %w", err)
-	}
-	status.APIAddress = apiAddress
-
 	oauthAddress, err := getRouteAddress(r, ctx, client.ObjectKeyFromObject(oauthRoute))
 	if err != nil {
 		return status, fmt.Errorf("failed get get route address: %w", err)
 	}
 	status.OAuthAddress = oauthAddress
 
-	vpnAddress, err := getLoadBalancerServiceAddress(r, ctx, client.ObjectKeyFromObject(vpnService))
+	apiAddress, err := getLoadBalancerServiceAddress(r, ctx, client.ObjectKeyFromObject(apiService))
 	if err != nil {
 		return status, fmt.Errorf("failed to get service: %w", err)
 	}
-	status.VPNAddress = vpnAddress
+	switch hcp.Spec.ControlPlaneServiceTypeStrategy {
+	case "NodePort":
+		status.APIAddress = hcp.Spec.ControlPlaneNodePortIngressTrafficDomain
+		status.APIPort = apiAddress
+	default:
+		status.APIAddress = apiAddress
+		status.APIPort = fmt.Sprint(APIServerPort)
+	}
+
+	vpnEnabled := true
```

Comment:
this seems hacky to me but thought I would try it for a protoype to show what I mean by "payload enablement/disablement"